charts/consul/templates/controller-deployment.yaml

```diff
@@ -1,5 +1,6 @@
 {{- if .Values.controller.enabled }}
 {{- if and .Values.global.adminPartitions.enabled (not .Values.global.enableConsulNamespaces) }}{{ fail "global.enableConsulNamespaces must be true if global.adminPartitions.enabled=true" }}{{ end }}
+{{- if and .Values.externalServers.enabled (not .Values.externalServers.hosts) }}{{ fail "externalServers.hosts must be set if externalServers.enabled is true" }}{{ end -}}
```

Comment:
I'm not going to update bats for controller in this PR because we'll combine this deployment with connect-inject deployment in a future PR before we go GA (NET-994)